File: cpp/contains_duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        if (nums.size() == 0) return false;
        bool result = false;
        sort(nums.begin(), nums.end());
        for (auto it=nums.begin(); it!=nums.end()-1;){
            if (*it == *++it){
                result = true;
                break;
            }
        }
        cout << result;
        return result;
    }
};

File: python/contains_duplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        sets = set()
        for n in nums:
            if n in sets:
                return True
            else:
                sets.add(n)
        return False
